src/components/probe/prober/index.ts

```diff
@@ -414,6 +417,95 @@ export class BaseProber implements Prober {
     log.warn(`${this.getMessagePrefix()} ${message.filter(Boolean).join(', ')}`)
   }
 
+  private initializeProbeState() {
+    if (
+      !this.probeConfig?.lastEvent ||
+      this.probeConfig.lastEvent?.recoveredAt !== null
+    ) {
+      return
+    }
+
+    if (!this.probeConfig.lastEvent?.alertId) {
+      log.error(`Last event ID in probe ${this.probeConfig.id} is required`)
+      return
+    }
+
+    const { alertId, createdAt } = this.probeConfig.lastEvent
+    const alert = this.getAlerts().find(({ id }) => id === alertId)
+
+    if (!alert) {
+      log.error(
+        `Alert ID: ${alertId} is not found in probe ${this.probeConfig.id}`
+      )
+      return
+    }
+
+    const request = this.getRequestByAlertId(alertId)
+    if (!request) {
+      log.error(
+        `Request for alert ID: ${alertId} is not found in probe ${this.probeConfig.id}`
+      )
+      return
+    }
+
+    startDowntimeCounter({
+      alert,
+      probeID: this.probeConfig.id,
+      url: request.url,
+      createdAt,
+    })
+  }
+
+  private getAlerts() {
+    const httpAlerts =
+      this.probeConfig?.requests?.map(({ alerts }) => alerts).find(Boolean) ||
+      []
+    const mariadbAlerts =
+      this.probeConfig?.mariadb?.map(({ alerts }) => alerts).find(Boolean) || []
+    const mongoAlerts =
+      this.probeConfig?.mongo?.map(({ alerts }) => alerts).find(Boolean) || []
+    const mysqlAlerts =
+      this.probeConfig?.mysql?.map(({ alerts }) => alerts).find(Boolean) || []
+    const pingAlerts =
+      this.probeConfig?.ping?.map(({ alerts }) => alerts).find(Boolean) || []
+    const postgresAlerts =
+      this.probeConfig?.postgres?.map(({ alerts }) => alerts).find(Boolean) ||
+      []
+    const redisAlerts =
+      this.probeConfig?.redis?.map(({ alerts }) => alerts).find(Boolean) || []
+    const socketAlerts = this.probeConfig?.socket?.alerts || []
+
+    return [
+      ...(this.probeConfig?.alerts || []),
+      ...httpAlerts,
+      ...mariadbAlerts,
+      ...mongoAlerts,
+      ...mysqlAlerts,
+      ...pingAlerts,
+      ...postgresAlerts,
+      ...redisAlerts,
+      ...socketAlerts,
+    ]
```

Comment:
**Suggestion:** Consider using optional chaining to simplify the code and avoid potential runtime errors.
```suggestion
  private getAlerts() {
    const httpAlerts = this.probeConfig?.requests?.flatMap(({ alerts }) => alerts) || []
    const mariadbAlerts = this.probeConfig?.mariadb?.flatMap(({ alerts }) => alerts) || []
    const mongoAlerts = this.probeConfig?.mongo?.flatMap(({ alerts }) => alerts) || []
    const mysqlAlerts = this.probeConfig?.mysql?.flatMap(({ alerts }) => alerts) || []
    const pingAlerts = this.probeConfig?.ping?.flatMap(({ alerts }) => alerts) || []
    const postgresAlerts = this.probeConfig?.postgres?.flatMap(({ alerts }) => alerts) || []
    const redisAlerts = this.probeConfig?.redis?.flatMap(({ alerts }) => alerts) || []
    const socketAlerts = this.probeConfig?.socket?.alerts || []

    return [
      ...(this.probeConfig?.alerts || []),
      ...httpAlerts,
      ...mariadbAlerts,
      ...mongoAlerts,
      ...mysqlAlerts,
      ...pingAlerts,
      ...postgresAlerts,
      ...redisAlerts,
      ...socketAlerts,
    ]
  }
```